src/Console/Shell.php

```diff
@@ -307,6 +310,7 @@ public function dispatchShell()
         if (is_string($args[0]) && count($args) === 1) {
             $args = explode(' ', $args[0]);
         }
+        $args[] = '--requested';
```

Comment:
The extra parameters could be merged into the arguments/params after the options are parsed. That would make the code inside a shell not need to know how it got its parameters.

If the parameters are not present in the option parser I don't think one could use them from the cli interface.